cfgov/legacy/housing_counselor/fetcher.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import absolute_import, unicode_literals
+
+import requests
+
+from six import print_
+
+
+HUD_COUNSELORS_URL = (
+    'https://data.hud.gov/Housing_Counselor/searchByLocation?'
+    'Lat=38.8951&Long=-77.0367&Distance=5000'
+)
+
+HUD_LANGUAGES_URL = 'https://data.hud.gov/Housing_Counselor/getLanguages'
+
+HUD_SERVICES_URL = 'https://data.hud.gov/Housing_Counselor/getServices'
+
+
+def fetch_counselors():
+    """Download housing counselor data from HUD.
+
+    This class fetches housing counselor data from data.hud.gov and also
+    expands any language and service abbreviations contained within.
+    """
+    counselors = download_housing_counselors(HUD_COUNSELORS_URL)
+
+    for key, url in (
+        ('languages', HUD_LANGUAGES_URL),
+        ('services', HUD_SERVICES_URL),
+    ):
+        replace_abbreviations(counselors, key, url)
+
+    return counselors
+
+
+def download_housing_counselors(url):
+    """Download HUD counselors from a given URL."""
+    print_('Downloading HUD counselors from', url, flush=True)
+    counselors = get_json_from_url(url)
+
+    if not counselors:
+        raise RuntimeError('Could not download HUD counselors')
+
+    print_('Retrieved', len(counselors), 'counselors', flush=True)
+    return counselors
+
+
+def replace_abbreviations(counselors, attribute, url):
+    """Replace attribute abbreviations with names from a given URL."""
+    print_('Downloading counselor', attribute, 'from', url, flush=True)
```

Comment:
Yeah, that's a good suggestion. I agree that using logging is probably better than using print like this. The reason I use this complicated `print_` to be honest is to get the `flush=True` behavior which isn't built into Python 2.7 yet, and this is future compatible with Python 3. I wanted to be able to flush after every statement so that slow-running jobs on e.g. Jenkins would update the output for review while they're still running.

That being said, the complication with using `logging` here instead would be that we don't want this code to log anything during unit tests, but we do when this code is running as part of a management command. I think because we don't have `LOGGING` defined in our `settings.test` I wouldn't have to do anything special to avoid unit test output, but I would then have to modify how this is run to get output on Jenkins. Currently there this is run using `DJANGO_SETTINGS_MODULE=cfgov.settings.test_nomigrations` to avoid migrations and avoid the need for MySQL. To enable logging I'd then need either another settings file or some kind of environment-variable based toggle.

Now that I write all of this out I think that's probably the better approach. Thanks for suggesting, I'll experiment with implementing that.